File: python/basic-calculator/src/calculator.ts
class Calculator {
    add(operand1: number, operand2: number): number {
        return operand1 + operand2;
    }

    subtract(operand1: number, operand2: number): number {
        return operand1 - operand2;
    }

    multiply(operand1: number, operand2: number): number {
        return operand1 * operand2;
    }

    divide(operand1: number, operand2: number): number {
        if (operand2 === 0) {
            throw new Error("Division by zero is not allowed.");
        }
        return operand1 / operand2;
    }
}

export default Calculator;

File: python/basic-calculator-1/src/calculator.py
def add(operand1, operand2):
    return operand1 + operand2

def subtract(operand1, operand2):
    return operand1 - operand2

def multiply(operand1, operand2):
    return operand1 * operand2

def divide(operand1, operand2):
    if operand2 == 0:
        raise ValueError("Cannot divide by zero.")
    return operand1 / operand2